shared/src/jvmMain/kotlin/org/jetbrains/kotlinconf/storage/ApplicationStorage.jvm.kt

```diff
@@ -0,0 +1,59 @@
+package org.jetbrains.kotlinconf.storage
+
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.launch
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.jetbrains.kotlinconf.ApplicationContext
+import java.io.File
+
+@Serializable
+private data class ApplicationStorageData(val values: Map<String, String>)
+
+@OptIn(FlowPreview::class, DelicateCoroutinesApi::class)
+actual fun ApplicationStorage(context: ApplicationContext): ApplicationStorage =
+    object : ApplicationStorage {
+        private val data: MutableStateFlow<ApplicationStorageData>
+
+        init {
+            data = try {
+                val content = File("store.json").readText()
+                val storageData = Json.decodeFromString<ApplicationStorageData>(content)
+                MutableStateFlow(storageData)
+            } catch (cause: Throwable) {
+                MutableStateFlow(ApplicationStorageData(emptyMap()))
+            }
+
+            GlobalScope.launch(Dispatchers.IO) {
+                data.debounce(5_000)
+                    .collect { storageData ->
+                    val content = Json.encodeToString(storageData)
+                    runCatching {
+                        File("store.json").writeText(content)
+                    }
+                }
```

Comment:
I'd say 5 sec is too much. And code formatting is broken here.
Could you extract file to the property and use standard app directory on user's home?

Reply:
I'm unsure if using the user directory is a good idea, cause we provide no way to uninstall it.